tests/test_api.py

```diff
@@ -12,15 +12,107 @@
 pkg_root = os.path.abspath(os.path.join(os.path.dirname(__file__), '..'))  # noqa
 sys.path.insert(0, pkg_root)  # noqa
 
-from dss.config import Replica
 import dss
 from dss.util import UrlBuilder
 from dss.util.version import datetime_to_version_format
-from tests.infra import DSSAssertMixin, DSSUploadMixin, DSSStorageMixin, TestBundle, testmode
+from dss.config import Replica, DeploymentStage
+from tests.infra import DSSAssertMixin, DSSUploadMixin, DSSStorageMixin, TestBundle, testmode, ExpectedErrorFields
 from tests.infra.server import ThreadedLocalServer
 from tests import get_auth_header
 
 
+class TestApiErrors(unittest.TestCase, DSSAssertMixin):
+    @classmethod
+    def setUpClass(cls):
+        cls.app = ThreadedLocalServer()
+        cls.app.start()
+        cls.app._chalice_app._override_exptime_seconds = 15.0
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.app.shutdown()
+
+    # all endpoints should return Retry-After headers for 50x responses, though we only test GET /bundle/{uuid} here
+
+    def test_retry_after_response_500_GET_bundle(self):
```

Comment:
this could get refactored into 2 base functions, then iterate through a set of possibilities, easier read.